test/hotspot/jtreg/runtime/cds/appcds/LambdaEagerInit.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+/*
+ * @test
+ * @bug 8257241
+ * @summary Run the LambdaEagerInitTest.java test in static CDS archive mode.
+ *          Create a base archive with the -Djdk.internal.lambda.disableEagerInitialization=true property.
+ *          Run with the archive with and without specifying the property.
+ *          With the disableEagerInit set to true, lambda proxy classes will not
+ *          be archived. During runtime, lambda proxy classes will not be loaded
+ *          from the archive.
+ * @requires vm.cds
+ * @library /test/lib /test/hotspot/jtreg/runtime/cds/appcds
+ * @compile ../../../../../jdk/java/lang/invoke/lambda/LambdaEagerInitTest.java
```

Comment:
I think it's better to make a copy of this test along side with this test.

Reply:
I can make a copy and put it under the `open/test/hotspot/jtreg/runtime/cds/appcds/test-classes` dir. I'd still need the `@compile` and would be something like `@compile test-classes/LambdaEagerInitTest.java`.

Reply:
The `@compile` can be avoiding by putting LambdaEagerInitTest into the `@library` and referencing it as

```
private static final String mainClass = LambdaEagerInitTest.class.getName();
```

Reply:
Yes, `@library test-classes` works.